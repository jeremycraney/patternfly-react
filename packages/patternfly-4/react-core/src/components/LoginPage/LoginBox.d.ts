import { SFC, HTMLProps, ReactNode } from 'react';

export interface LoginBoxProps extends HTMLProps<HTMLDivElement> {
  children?: ReactNode;
  className?: string;
}

declare const LoginBox: SFC<LoginBoxProps>;

export default LoginBox;
