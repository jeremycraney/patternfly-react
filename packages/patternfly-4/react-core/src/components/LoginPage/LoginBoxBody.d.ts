import { SFC, HTMLProps, ReactNode } from 'react';

export interface LoginBoxBodyProps extends HTMLProps<HTMLDivElement> {
  children?: ReactNode;
  className?: string;
}

declare const LoginBoxBody: SFC<LoginBoxBodyProps>;

export default LoginBoxBody;
