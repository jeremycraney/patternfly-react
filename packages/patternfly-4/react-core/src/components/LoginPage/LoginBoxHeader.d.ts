import { SFC, HTMLProps, ReactNode } from 'react';

export interface LoginBoxHeaderProps extends HTMLProps<HTMLDivElement> {
  children?: ReactNode;
  className?: string;
  dropdown?: ReactNode;
  title?: string;
  subtitle?: string;
}

declare const LoginBoxHeader: SFC<LoginBoxHeaderProps>;

export default LoginBoxHeader;
