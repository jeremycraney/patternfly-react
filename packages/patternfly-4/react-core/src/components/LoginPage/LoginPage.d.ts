import { SFC, HTMLProps, ReactNode } from 'react';
import { OneOf } from '../../typeUtils';

export const LoginListVariant: {
  grid: 'grid';
  inline: 'inline';
};

export interface LoginPageProps extends HTMLProps<HTMLElement> {
  children?: ReactNode;
  className?: string;
  mainBrandImgSrc?: string;
  mainBrandImgAlt?: string;
  mainBackgroundImgSrc?: string;
  mainBackgroundImgAlt?: string;
  mainTextContent?: string;
  mainListItems?: ReactNode;
  mainListVariants?: OneOf<typeof LoginListVariant, keyof typeof LoginListVariant>;
  loginTitle: string;
  loginSubtitle?: ReactNode;
  loginLanguageDropdown?: ReactNode;
  loginFooterContent?: ReactNode;
}

declare const LoginPage: SFC<LoginPageProps>;

export default LoginPage;
