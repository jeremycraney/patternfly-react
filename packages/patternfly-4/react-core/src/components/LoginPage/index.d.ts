export { default as LoginPage, LoginPageProps } from './LoginPage';
