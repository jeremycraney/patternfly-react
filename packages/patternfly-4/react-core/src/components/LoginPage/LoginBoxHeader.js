import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from '@patternfly/react-styles';
import { Title } from '../Title';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/patternfly-next/components/LoginBox/login-box.css';
import { global_spacer_md, global_spacer_sm } from '@patternfly/react-tokens';
import displayStyles from '@patternfly/patternfly-next/utilities/Display/display.css';
import flexAlignment from '@patternfly/patternfly-next/utilities/Flex/flex.css';

const propTypes = {
  /** Content rendered inside the LoginBox Header */
  children: PropTypes.node,
  /** Additional classes added to the LoginBox Header */
  className: PropTypes.string,
  /** Dropdown component for the LoginBox Header */
  dropdown: PropTypes.node,
  /** Title for the LoginBox Header */
  title: PropTypes.string,
  /** Subtitle that contains the Text, URL, and URL Text for the LoginBox Header */
  subtitle: PropTypes.node
};

const defaultProps = {
  children: null,
  className: '',
  dropdown: null,
  title: '',
  subtitle: ''
};

const tokenStyles = StyleSheet.create({
  title: {
    marginBottom: global_spacer_sm.var
  },
  text: {
    marginLeft: global_spacer_md.var
  }
});

const LoginBoxHeader = ({ children, className, dropdown, title, subtitle, ...props }) => (
  <div className={css(styles.loginBoxHeader, className)} {...props}>
    <Title className={css(tokenStyles.title)} size="3xl">
      {title}
    </Title>
    <div className={css(displayStyles.displayInlineFlex, flexAlignment.alignItemsCenter)}>
      {dropdown}
      {subtitle && <p className={css(tokenStyles.text)}>{subtitle}</p>}
    </div>

    {children}
  </div>
);

LoginBoxHeader.propTypes = propTypes;
LoginBoxHeader.defaultProps = defaultProps;

export default LoginBoxHeader;
