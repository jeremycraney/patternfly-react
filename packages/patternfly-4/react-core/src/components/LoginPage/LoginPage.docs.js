import { LoginPage } from '@patternfly/react-core';
import Simple from './examples/SimpleLoginPage';

export default {
  title: 'LoginPage',
  components: {
    LoginPage
  },
  examples: [Simple],
  fullPageOnly: true
};
