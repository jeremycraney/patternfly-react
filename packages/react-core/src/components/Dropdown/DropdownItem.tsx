import React from 'react';
import { css } from '@patternfly/react-styles';
import { MenuItemProps, MenuItem } from '../Menu';
import { useOUIAProps, OUIAProps } from '../../helpers';

export interface DropdownItemProps extends Omit<MenuItemProps, 'ref'>, OUIAProps {
  /** Anything which can be rendered in a dropdown item */
  children?: React.ReactNode;
  /** Classes applied to root element of dropdown item */
  className?: string;
  /** Description of the dropdown item */
  description?: React.ReactNode;
  /** Render item as disabled option */
  isDisabled?: boolean;
  /** Identifies the component in the dropdown onSelect callback */
  itemId?: any;
  /** Callback for item click */
  onClick?: (event?: any) => void;
  /** Value to overwrite the randomly generated data-ouia-component-id.*/
  ouiaId?: number | string;
  /** Set the value of data-ouia-safe. Only set to true when the component is in a static state, i.e. no animations are occurring. At all other times, this value must be false. */
  ouiaSafe?: boolean;
}

export const DropdownItem: React.FunctionComponent<MenuItemProps> = ({
  children,
  className,
  description,
  isDisabled,
  itemId,
  onClick,
  ouiaId,
  ouiaSafe,
  ...props
}: DropdownItemProps) => {
  const ouiaProps = useOUIAProps(DropdownItem.displayName, ouiaId, ouiaSafe);
  return (
    <MenuItem
      className={css(className)}
      description={description}
      isDisabled={isDisabled}
      itemId={itemId}
      onClick={onClick}
      {...ouiaProps}
      {...props}
    >
      {children}
    </MenuItem>
  );
};
DropdownItem.displayName = 'DropdownItem';
