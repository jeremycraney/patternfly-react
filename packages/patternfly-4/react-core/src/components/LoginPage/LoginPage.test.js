import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from './LoginPage';
import { Dropdown, DropdownItem } from '../Dropdown';

const dropdownToggle = jest.fn();

const myDropdown = (
  <Dropdown toggle={dropdownToggle}>
    <DropdownItem>English</DropdownItem>
  </Dropdown>
);

const subtitle = (
  <React.Fragment>
    Login to your account <a href="https://www.patternfly.org">Need an account?</a>
  </React.Fragment>
);

const props = {
  mainListVariants: 'inline',
  mainBrandImgSrc: 'Brand src',
  mainBrandImgAlt: 'Pf-logo',
  mainBackgroundImgSrc: 'Background src',
  mainBackgroundImgAlt: 'Pf-background',
  mainListItems: 'English',
  mainTextContent: 'This is placeholder text only.',
  loginTitle: 'Log into your account',
  loginSubtitle: subtitle,
  loginLanguageDropdown: myDropdown,
  loginFooterContent: 'Footer'
};

test('check loginpage example against snapshot', () => {
  const view = shallow(<LoginPage {...props} />);
  expect(view).toMatchSnapshot();
});
