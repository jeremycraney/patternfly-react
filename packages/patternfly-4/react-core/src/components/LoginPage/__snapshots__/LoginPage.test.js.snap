// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check loginpage example against snapshot 1`] = `
<React.Fragment>
  <BackgroundImage
    alt="Pf-background"
    className=""
    src="Background src"
  />
  <Login
    className=""
    footer={
      <LoginFooter
        className=""
      >
        <List
          className=""
          variant="inline"
        >
          English
        </List>
      </LoginFooter>
    }
    header={
      <LoginHeader
        className=""
        headerBrand={
          <React.Fragment>
            <LoginHeaderBrand
              className=""
            >
              <Brand
                alt="Pf-logo"
                className=""
                src="Brand src"
              />
            </LoginHeaderBrand>
            <TextContent
              className=""
            >
              <Text
                className=""
                component="p"
              >
                This is placeholder text only.
              </Text>
            </TextContent>
          </React.Fragment>
        }
      />
    }
  >
    <LoginBox
      className=""
    >
      <LoginBoxHeader
        className=""
        dropdown={
          <Dropdown
            className=""
            direction="down"
            dropdownItems={Array []}
            isOpen={false}
            isPlain={false}
            onSelect={[Function]}
            position="left"
            toggle={[MockFunction]}
          >
            <Item
              className=""
              component="a"
              href="#"
              isDisabled={false}
              isHovered={false}
            >
              English
            </Item>
          </Dropdown>
        }
        subtitle={
          <React.Fragment>
            Login to your account 
            <a
              href="https://www.patternfly.org"
            >
              Need an account?
            </a>
          </React.Fragment>
        }
        title="Log into your account"
      />
      <LoginBoxBody
        className=""
      />
      <LoginBoxFooter
        className=""
      >
        Footer
      </LoginBoxFooter>
    </LoginBox>
  </Login>
</React.Fragment>
`;
